Storage/ThirdParty/SwiftData.swift

```diff
@@ -87,7 +89,14 @@ class DeferredDBOperation<T>: Deferred<T>, Cancellable {
     }
     
     override func fill(_ value: T) {
-        dispatchWorkItem = nil
+        defer {
+            dispatchWorkItem = nil
+        }
+
+        guard !cancelled else {
```

Comment:
I prefer removing this and calling fill() and having the caller check `result.cancelled`, because that seems more consistent with what I would expect as a consumer of this API. However, I don't want to bikeshed this because this cancellation is used in *one* place in the entire code :).